client/cli/src/lib.rs

```diff
@@ -416,59 +460,19 @@ enum Command {
 
     /// Gets the current epoch count
     EpochCount,
-}
 
-/// Extension trait for values that either return `T` or terminate the program
-/// with CLI error
-trait OrTerminate<T> {
-    fn or_terminate(self, err: CliErrorKind, msg: &str) -> T;
-}
-
-trait ErrorHandler<T, E>: OrTerminate<T> {
-    fn transform<F>(self, success: F, err: CliErrorKind, msg: &str) -> CliResult
-    where
-        F: Fn(T) -> CliOutput;
-}
+    /// Call module-specific commands
+    Module {
+        id: String,
 
-impl<T, E: Into<Box<dyn Error>>> OrTerminate<T> for Result<T, E> {
-    fn or_terminate(self, err: CliErrorKind, msg: &str) -> T {
-        match self {
-            Ok(v) => v,
-            Err(e) => {
-                let cli_error = CliError::from(err, msg, Some(e.into()));
-                eprintln!("{cli_error}");
-                exit(1);
-            }
-        }
-    }
-}
-
-impl<T, E: Into<Box<dyn Error>>> ErrorHandler<T, E> for Result<T, E> {
-    fn transform<F>(self, success: F, err: CliErrorKind, msg: &str) -> CliResult
-    where
-        F: Fn(T) -> CliOutput,
-    {
-        match self {
-            Ok(v) => Ok(success(v)),
-            Err(e) => Err(CliError::from(err, msg, Some(e.into()))),
-        }
-    }
+        /// Command with arguments to call the module with
+        arg: Vec<ffi::OsString>,
+    },
 }
 
-impl<T> OrTerminate<T> for Option<T> {
-    fn or_terminate(self, err: CliErrorKind, msg: &str) -> T {
-        match self {
-            Some(v) => v,
-            None => {
-                let cli_error = CliError::from(err, msg, None);
-                eprintln!("{cli_error}");
-                exit(1);
-            }
-        }
-    }
-}
+type CliResult<E> = Result<E, CliError>;
 
-type CliResult = Result<CliOutput, CliError>;
+type CliOutputResult = Result<CliOutput, CliError>;
```

Comment:
Maybe move this and `CliResult` up by `CliError` up by `CliError` and add comments also.